libpod/image/image.go

```diff
@@ -156,6 +156,41 @@ func (ir *Runtime) New(ctx context.Context, name, signaturePolicyPath, authfile
 	return &newImage, nil
 }
 
+// NewFromLoad creates a new image object for images pulled from a tar archive (podman load)
+// This function is needed because it is possible for a tar archive to have multiple tags for one image
+func (ir *Runtime) NewFromLoad(ctx context.Context, name, signaturePolicyPath string, writer io.Writer) ([]*Image, error) {
+	var newImages []*Image
+	newImage := Image{
+		InputName:    name,
+		Local:        false,
+		imageruntime: ir,
+	}
+
+	if signaturePolicyPath == "" {
+		signaturePolicyPath = ir.SignaturePolicyPath
+	}
+	imageNames, err := newImage.pullImage(ctx, writer, "", signaturePolicyPath, SigningOptions{}, &DockerRegistryOptions{}, false)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to pull %s", name)
+	}
+
+	for _, name := range imageNames {
+		newImage := Image{
+			InputName:    name,
+			Local:        false,
+			imageruntime: ir,
+		}
+		img, err := newImage.getLocalImage()
+		if err != nil {
+			return nil, errors.Wrapf(err, "error retrieving local image after pulling %s", name)
+		}
+		newImage.image = img
+		newImages = append(newImages, &newImage)
+	}
+
+	return newImages, nil
```

Comment:
Instead of returning an array of images, can we make a single Image hold enough information that we can identify that all the tags belong to it?

Reply:
I am not sure. I know we will be pulling multiple times for different tags (just the layers that are diff), so I guess not.